File: code/js/prototypes/location.js
Location.prototype = new DataType();
Location.prototype.constructor = Location;

function Location(LID, description, number, street, city, postalcode, country, user_UID) {
	this.LID = LID;
	this.description = description;
	this.number = number;
	this.street = street;
	this.city = city;
	this.postalcode = postalcode;
	this.country = country;
	this.user_UID = user_UID;
}

File: code/model/location.py

from sparrow import *

from .owentity import *
from .user import User

class Location(RTOwEntity):
    key = LID = KeyProperty()
    description = Property(str, constraint=lambda d: len(d) <= 500)
    
    # address
    number = Property(int)
    street = Property(str)
    city = Property(str, constraint=lambda d: len(d) <= 200)
    postalcode = Property(int)
    country = Property(str, constraint=lambda d: len(d) <= 100)
    
    user = Reference(User)
    
    def is_authorized(self, type, usr):
        return usr.key == self.user
